nav2_behavior_tree/plugins/decorator/single_trigger_node.cpp

```diff
@@ -37,11 +37,24 @@ BT::NodeStatus SingleTrigger::tick()
   setStatus(BT::NodeStatus::RUNNING);
 
   if (first_time_) {
-    first_time_ = false;
-    return BT::NodeStatus::SUCCESS;
-  } else {
-    return BT::NodeStatus::FAILURE;
+    const BT::NodeStatus child_state = child_node_->executeTick();
+
+    switch (child_state) {
+      case BT::NodeStatus::RUNNING:
+        return BT::NodeStatus::RUNNING;
+
+      case BT::NodeStatus::SUCCESS:
+        first_time_ = false;
+        return BT::NodeStatus::SUCCESS;
+
+      case BT::NodeStatus::FAILURE:
```

Comment:
I think even failure `first_time_` should be reset to `false`, since this node implies we only trigger this child once, regardless of the return code

Reply:
e.g. mimic the `default` below

Reply:
The switch case moves to the default case when case is `BT::NodeStatus::FAILURE`, no need to mimic the default again.

Reply:
https://stackoverflow.com/questions/38218812/c-empty-case-before-default-useless

I tend to agree with the first commenter, this looks incomplete and has some implicit flow requiring the reader to me more than average knowledgeable about how `case/switch` works in C++ and easily scanned over incorrectly. There's no functional difference between these 2 sets of code, but sets explicit boundaries about what's happening.

Reply:
Yup, makes sense